src/inflators/spot-light.ts

```diff
@@ -0,0 +1,53 @@
+import { addComponent } from "bitecs";
+import { addObject3DComponent } from "../utils/jsx-entity";
+import { SpotLightTag, LightTag } from "../bit-components";
+import { SpotLight } from "three";
+import { HubsWorld } from "../app";
+
+export type SpotLightParams = {
+  color: string;
+  intensity: number;
+  range: number;
+  decay: number;
+  innerConeAngle: number;
+  outerConeAngle: number;
+  castShadow: boolean;
+  shadowMapResolution: [number, number];
+  shadowBias: number;
+  shadowRadius: number;
+};
+
+const DEFAULTS = {
```

Comment:
Not sure if I like it better but I notice we have to duplicate a lot of information with these defaults... 

We could use typeof: [example](https://www.typescriptlang.org/play?#code/MYewdgzgLgBAIgUQGIEECqAZAKgZRgXhgG8BYAKBhgEswoBTSKqATwC4YBGAOgAYAacpQBOAQzABzOu36CYAEzrARbGACZeAitTBg6QgMLg6KCQBspMGVpABXegaMnx59gFkRUABZcACgEkYAHoYABYNWSVoHE8RORAAd3YoIRs6TUoIGLj49wAHACU6CBBTOypwdgBtAFYOVT4YWtUAXRgRCBhK+Ko5L1YwGwBbACM9Bs86KnFPKH6h0aFm9JhM2ISAISp26WXV7PzYqhsIdm4ecgBfAG5ycjoAD1yQIVgWXLoYHCeoDCmZnxEokGHUIbzoIAAZvBkOhsDgbmQgA)